spec/index.bs

```diff
@@ -2517,6 +2524,52 @@ dictionary GPUImageBitmapCopyView {
     </div>
 </div>
 
+### <dfn method for=GPUCommandEncoder>inlineUpdateBuffer(source, sourceOffset, destination, destinationOffset, size)</dfn> ### {#GPUCommandEncoder-inlineUpdateBuffer}
+
+It's often useful for applications to update buffer data prior to draw or compute operations.
+For example, updating model-view and projection matrices before or interleaved with rendering of a scene.
+When these uploads are small, it's viable to inline the update data into the command buffer.
+This does require more copies than other upload paths, but for small data sizes this overhead is negligible.
+Implementations are expected to warn against using this for medium-to-large buffer updates. (e.g. >64k)
+
+In Vulkan, this is similar to |vkCmdUpdateBuffer|.
+In D3D12, implementations can leverage |ID3D12GraphicsCommandList2::WriteBufferImmediate|.
+Metal might use |makeBuffer(bytesNoCopy:length:options:deallocator:)| around some section of shared command buffer serialization memory.
```

Comment:
I wonder how this would work in practice. It requires page-size alignment for both the pointer and the size, and also:
> The existing memory allocation must be covered by a single VM region, typically allocated with vm_allocate or mmap. Memory allocated by malloc is specifically disallowed.